packages/flutter/test/widgets/async_test.dart

```diff
@@ -66,6 +66,19 @@ void main() {
       ));
       expect(find.text('AsyncSnapshot(ConnectionState.waiting, null, null)'), findsOneWidget);
     });
+    testWidgets('gracefully handles transition from null future with initial data without introducing new states', (WidgetTester tester) async {
+      final String initial = 'initial';
```

Comment:
As a matter of personal preference (feel free to ignore), I find it easier to read a short test case that uses literal data rather than variables which have to be declared and interpolated into strings. Even if that literal has to be repeated once or twice. Some of the later test cases use `'A'` and `'B'` in this manner.

I'd probably just use `'I'` for the initial data.